examples/Backtesting/custom_analytical.py

```diff
@@ -1,124 +1,110 @@
-from baybe.parameters import NumericalDiscreteParameter
-from baybe.recommenders import (
-    BotorchRecommender,
-    RandomRecommender,
-    TwoPhaseMetaRecommender,
-)
+from baybe.parameters.numerical import NumericalContinuousParameter
+from baybe.recommenders import RandomRecommender
 from baybe.searchspace import SearchSpace
-from baybe.simulation import simulate_scenarios
+from baybe.simulation import label_columns, simulate_scenarios
 from baybe.targets import NumericalTarget
+from baybe.utils.plotting import create_example_plots
 
-### Parameters for a full simulation loop
+### Settings
 
-# For the full simulation, we need to define some additional parameters.
-# These are the number of Monte Carlo runs and the number of experiments to be conducted per run.
-
-# The parameter `POINTS_PER_DIM` controls the number of points per dimension.
-# Note that the searchspace will have `POINTS_PER_DIM**DIMENSION` many points.
+# Before we start, let us collect a few general settings for the example:
 
 SMOKE_TEST = "SMOKE_TEST" in os.environ
 
-N_MC_ITERATIONS = 2 if SMOKE_TEST else 5
-N_DOE_ITERATIONS = 2 if SMOKE_TEST else 5
-DIMENSION = 4
-BOUNDS = [(-2, 2), (-2, 2), (-2, 2), (-2, 2)]
-POINTS_PER_DIM = 3 if SMOKE_TEST else 10
-
-### Defining the test function
+BATCH_SIZE = 1
+N_MC_ITERATIONS = 2 if SMOKE_TEST else 20
+N_DOE_ITERATIONS = 2 if SMOKE_TEST else 30
+DIMENSION = 1 if SMOKE_TEST else 10
+BOUNDS = (-1, 1)
 
-# See [here](./../Searchspaces/continuous_space_custom_function.md) for details.
+### Defining the Optimization Problem
 
+# Now, we can define the scope of our optimization problem. Our goal is to optimize
+# a high-dimensional quadratic function on a bounded input domain. We first define
+# the corresponding inputs and output of the function:
 
-def sum_of_squares(*x: float) -> float:
-    """Calculate the sum of squares."""
-    res = 0
-    for y in x:
-        res += y**2
-    return res
+parameters = [
+    NumericalContinuousParameter(name=f"x_{k}", bounds=BOUNDS) for k in range(DIMENSION)
+]
+target = NumericalTarget(name="Target", mode="MIN")
 
 
-### Creating the searchspace and the objective
+# Based on the above, we construct the black-box callable to be optimized.
+# Using the {func}`~baybe.simulation.lookup.label_columns` decorator, we can easily map
+# the columns of the raw input/output arrays to our parameter and target objects, which
+# creates the required dataframe-based lookup for the optimization loop:
 
-# As we expect it to be the most common use case, we construct a purely discrete space here.
-# Details on how to adjust this for other spaces can be found in the searchspace examples.
 
-parameters = [
-    NumericalDiscreteParameter(
-        name=f"x_{k+1}",
-        values=list(np.linspace(*BOUNDS[k], POINTS_PER_DIM)),
-        tolerance=0.01,
-    )
-    for k in range(DIMENSION)
-]
+@label_columns([p.name for p in parameters], [target.name])
+def sum_of_squares(x: np.ndarray, /) -> np.ndarray:
```

Comment:
As a user, I would be confused here: The `calleble` needs to map `DataFrame` to `DataFrame`, but here we talk about `np.ndarray`? I would actually prefer the example to contain a Callable having exactly that signature, and then providing the alternative using the decorator.